src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -1420,7 +1429,35 @@ internal void RegisterIncludes(BlittableJsonReaderObject includes)
                 IncludedDocumentsById[newDocumentInfo.Id] = newDocumentInfo;
             }
         }
+        
+        internal void RegisterRevisionIncludes(BlittableJsonReaderObject includes)
+        {
+            if (NoTracking)
+                return;
+
+            if (includes == null)
+                return;
+
+            IncludedRevisionByChangeVectors ??= new Dictionary<string, DocumentInfo>(StringComparer.OrdinalIgnoreCase);
+            var propertyDetails = new BlittableJsonReaderObject.PropertyDetails();
+            for (int i = 0; i < includes.Count; i++)
+            {
+                includes.GetPropertyByIndex(i, ref propertyDetails);
+
+                if (propertyDetails.Value == null)
+                    continue;
 
+                var json = (BlittableJsonReaderObject)propertyDetails.Value;
+
+                var newDocumentInfo = DocumentInfo.GetNewDocumentInfo(json);
+                if (newDocumentInfo.Metadata.TryGetConflict(out var conflict) && conflict)
+                    continue;
+                if (newDocumentInfo.Metadata.TryGet(Constants.Documents.Metadata.ChangeVector,out string cv))
```

Comment:
Is there a chance that there will be no CV? Maybe we should throw then?

Reply:
I think the best idea is to be consistent as you said. compare to the `include document` we return rather than throwing an error.
https://github.com/ravendb/ravendb/pull/12242/files#r648209360